### ISSUE TITLE
Corrupted 16x16 icon.ico when using png

### ISSUE THREAD

**Issue by nothingismagick** [idesis-gmbh/png2icons#4]
I am using a buffer passed from sharp. Generally png2icons works fine, except when I try to make a png-based icon.ico file.
```
sharp(src)
  .resize(256, 256)
  .crop(sharp.strategy.centre)
  .png()
  .toBuffer()
  .then(data => {
    let out = png2icons.createICO(data, png2icons.BICUBIC, 0, true)
    fs.writeFileSync(`${target}/electron/icon.ico`, out)
  })
  .catch(err => console.log(err))
```
I expected to see something like this (which is the result of a bmp not png type):
![screen shot 2018-05-16 at 00 52 39](https://user-images.githubusercontent.com/35242872/40087589-825647a4-58a3-11e8-8205-827e0c3f3b67.png)

Instead this is what I got<br>
![screen shot 2018-05-16 at 00 55 17](https://user-images.githubusercontent.com/35242872/40087665-d822f448-58a3-11e8-85ed-4e839701b7f4.png)

When I change the number of colors to e.g. 256, this problem disappears (but I am unsatisfied with the banding in gradients.)

**Comment by mn4367** [comment]
Can you upload the PNG file (and maybe also the original file) somewhere? I'm not sure if I understood your problem correctly, if you pass a buffer then it works but it fails when you load the buffer from a file?

**Comment by nothingismagick** [comment]
I updated my issue (sorry - I wrote it quite late at night :| ).
Here is a link to the .ico and a link to the .png
- [ico](https://github.com/nothingismagick/icon-factory/blob/master/test/sources/broken_icon.ico)
- [original](https://github.com/nothingismagick/icon-factory/blob/master/test/example-1240x1240.png)

**Comment by mn4367** [comment]
Hmm, using this command line 

```
node png2icons-cli.js example-1240x1240.png example-1240x1240 -ico -bz -i
```

with your example file this result is produced, which seems to be OK:

![bildschirmfoto 2018-05-16 um 14 14 47](https://user-images.githubusercontent.com/6379696/40116843-45870f9e-5915-11e8-8550-d54ef167cda4.png)

My suspicion is that something around `toBuffer()` produces wrong data. `png2icons.createICO` expects a complete PNG "file" as input (not only pixel data). In your case it means that `data` must contain the same bytes as the corresponding PNG file. If you save the `data` buffer to a file, is that file a valid PNG? In other words, if you process this file with the CLI, are the same artifacts visible?

**Comment by nothingismagick** [comment]
Yes, here is the command I am using with the same buffer to make an icns file, which works exactly as expected (from the buffer):

```
sharp(src)
.resize(256, 256)
.crop(sharp.strategy.centre)
.png()
.toBuffer()
.then(data => {
	png2icons.setLogger(console.log)
	let out = png2icons.createICNS(data, png2icons.BICUBIC, 0)
	fs.writeFileSync(`${target}/electron/icon.icns`, out)
}
```
I also know for a fact that the example file also creates a totally valid (albeit larger) icns set and ico set (when I use the `var input = fs.readFileSync("sample.png")`approach...)

**Comment by mn4367** [comment]
Does the problem still exist? I'm completely unsure what I could do since I can't reproduce the problem here.

**Comment by mn4367** [comment]
Fixed with v2.0.0.
